### PR TITLE
feat: library

### PR DESCRIPTION
**Pull request by CurryPseudo**
Also fix current example scheme file and interpreter test with standard library import

### CODE REVIEW

File: examples/library.scm [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+            (display "Hello, ")
+            (display who)
+            (display "!")
+            (newline)
+        )
+    )
+)
+(import (foo))
+(import (example bar))
+(display we-are-who)
```
Comment (Danielmelody) [inline examples/library.scm:15]: should append an extra line

**Comment by CurryPseudo** [comment]
updated

File: src/error.rs [hunk 2]
```diff
@@ -40,6 +34,17 @@ pub trait ToLocated {
...
     }
 }
+impl<T: PartialEq> Located<T> {
+    pub fn map<U: PartialEq>(self, f: impl Fn(T) -> U) -> Located<U> {
+        Located::<U> {
+            data: f(self.data),
+            location: self.location,
+        }
+    }
+    pub fn into_data(self) -> T {
```
Comment (Danielmelody) [inline src/error.rs:44]: Maybe named as `extract_data`?

File: src/parser/parser.rs [hunk 3]
```diff
@@ -231,10 +273,102 @@ impl<TokenIter: Iterator<Item = Result<Token>>> Parser<TokenIter> {
...
+                        self.advance(1)?;
+                        return Ok(Located::from(ExportSpec::Rename(from, to)));
+                    }
+                    _ => (),
+                }
+            }
+            Some(TokenData::Identifier(identifier)) => {
+                return Ok(Located::from(ExportSpec::Direct(identifier)));
+            }
+            _ => (),
```
Comment (Danielmelody) [inline src/parser/parser.rs:365]: I think the error should be wrote here.

File: src/parser/parser.rs [hunk 4]
```diff
@@ -231,10 +273,102 @@ impl<TokenIter: Iterator<Item = Result<Token>>> Parser<TokenIter> {
...
+                match self.current_identifier()?.as_str() {
+                    "rename" => {
+                        self.advance(1)?;
+                        let from = self.current_identifier()?;
+                        self.advance(1)?;
+                        let to = self.current_identifier()?;
+                        self.advance(1)?;
+                        return Ok(Located::from(ExportSpec::Rename(from, to)));
+                    }
+                    _ => (),
```
Comment (Danielmelody) [inline src/parser/parser.rs:359]: You can write expect_nth_token to avoid this arm.

File: src/parser/parser.rs [hunk 5]
```diff
@@ -440,24 +576,55 @@ impl<TokenIter: Iterator<Item = Result<Token>>> Parser<TokenIter> {
...
+                    return Ok(LibraryNameElement::Identifier(identifier).locate(self.location))
+                }
+                TokenData::Primitive(primitive) => match primitive {
+                    Primitive::Integer(i) => {
+                        if i >= 0 {
+                            return Ok(LibraryNameElement::Integer(i as u32).locate(self.location));
+                        } else {
+                        }
+                    }
+                    _ => (),
```
Comment (Danielmelody) [inline src/parser/parser.rs:592]: And all those xxx =>  (), should be avoid.

**Pull request merged**